lib/rules/no-shadow.js

```diff
@@ -165,6 +165,81 @@ module.exports = {
                     !isOnInitializer(variable, shadowed) &&
                     !(options.hoist !== "all" && isInTdz(variable, shadowed))
                 ) {
+                    if (shadowed.defs[0]) {
+                        const shadowedNode = shadowed.defs[0].node;
+                        const shadowNode = variable.defs[0].node;
+
+                        // Skip variable declaration
+                        if (
```

Comment:
Instead of hardcoding this, can we instead traverse up the tree using a loop and checking the `parent` property of the node? We'll need to be careful for situations like the following, though:

```js

const b = [].filter(/* not shadowed */ b => b.find(/* shadowed */ b => b > 10))
```

Reply:
I have added the requested test case.
I've also noticed that the example you've given,
`const b = [].filter(/* not shadowed */ b => b.find(/* shadowed */ b => b > 10))`
was not raising no-shadow error, so I've also committed a fix for that, as well as adding the example as a test case.

I'm new with working on es-lint so I'm not entirely sure how to avoid hardcoding to solve this issue.